client/state/comments/actions.js

```diff
@@ -0,0 +1,303 @@
+ * @returns {Function} a thunk that creates a comment for a given post
+ */
+export function writeComment( commentText, siteId, postId, parentCommentId ) {
+	if ( ! commentText || ! siteId || ! postId ) {
+		return;
+	}
+
+	return ( dispatch ) => {
+		const placeholderComment = createPlaceholderComment( commentText, postId, parentCommentId );
+
+		// Insert a placeholder
+		dispatch( {
+			type: COMMENTS_RECEIVE,
+			siteId,
+			postId,
+			comments: [ placeholderComment ]
+		} );
+
+		let apiPromise;
+
+		if ( parentCommentId ) {
+			apiPromise = wpcom.site( siteId ).post( postId ).comment( parentCommentId ).reply( {}, commentText );
+		} else {
+			apiPromise = wpcom.site( siteId ).post( postId ).comment().add( commentText );
+		}
+
+		return apiPromise.then( ( comment ) => {
+			// remove the placeholder
+			dispatch( {
+				type: COMMENTS_REMOVE,
+				siteId,
+				postId,
+				commentId: placeholderComment.ID
+			} );
+
+			// insert the real comment
+			dispatch( {
+				type: COMMENTS_RECEIVE,
+				siteId,
+				postId,
+				comments: [ comment ]
+			} );
+
+			return comment;
+		} )
+		.catch( ( error ) => {
+			dispatch( {
+				type: COMMENTS_ERROR,
+				siteId,
+				postId,
+				commentId: placeholderComment.ID,
+				error
+			} );
+
+			throw error;
+		} );
+	};
+}
+
+/***
+ * Creates a thunk that likes a comment
+ * @param {Number} siteId site identifier
+ * @param {Number} postId post identifier
+ * @param {Number} commentId comment identifier
+ * @returns {Function} think that likes a comment
+ */
+export function likeComment( siteId, postId, commentId ) {
+	return ( dispatch ) => {
+		// optimistic update
+		dispatch( {
+			type: COMMENTS_LIKE,
+			siteId,
+			postId,
+			commentId
+		} );
+
+		// optimistic revert on error, return here for test more conveniently
+		return wpcom.site( siteId ).comment( commentId ).like().add( { source: 'reader' } ).then( ( data ) => dispatch( {
+			type: COMMENTS_LIKE_UPDATE,
+			siteId,
+			postId,
+			commentId,
+			iLike: data.i_like,
+			likeCount: data.like_count
+		} ) ).catch( () => dispatch( {
+			type: COMMENTS_UNLIKE,
+			siteId,
+			postId,
+			commentId
+		} ) );
+	};
+}
+
+/***
+ * Creates a thunk that unlikes a comment
+ * @param {Number} siteId site identifier
+ * @param {Number} postId post identifier
+ * @param {Number} commentId comment identifier
+ * @returns {Function} think that unlikes a comment
+ */
+export function unlikeComment( siteId, postId, commentId ) {
+	return ( dispatch ) => {
+		// optimistic update
+		dispatch( {
+			type: COMMENTS_UNLIKE,
+			siteId,
+			postId,
+			commentId
+		} );
+
+		// optimistic revert on error, return here for test more conveniently
+		return wpcom.site( siteId ).comment( commentId ).like().del( { source: 'reader' } ).then( ( data ) => dispatch( {
+			type: COMMENTS_LIKE_UPDATE,
+			siteId,
+			postId,
+			commentId,
+			iLike: data.i_like,
+			likeCount: data.like_count
+		} ) ).catch( () => dispatch( {
```

Comment:
I don't think that actually fetches, it just returns an object that can be used to chain further API things. If it fetched, it would have to take a callback or hand back a Promise, which it doesn't appear to do.

We may need to validate that an successful like falls through into catch though. Separately, try liking a comment that  doesn't exist to see how it responds.